tools/rust_analyzer/lib.rs

```diff
@@ -61,15 +59,10 @@ pub fn write_rust_project(
         rules_rust_name.as_ref(),
     )?;
 
-    let workspace_name = match rules_rust_name.as_ref().trim_start_matches('@') {
-        "" => SYSROOT_SRC_FILE_RUNFILES_PREFIX,
-        s => s,
-    };
-    let toolchain_info_path = format!(
-        "{workspace_name}/rust/private/rust_analyzer_detect_sysroot.rust_analyzer_toolchain.json"
+    let path = runfiles::rlocation!(
+        Runfiles::create()?,
+        "rules_rust/rust/private/rust_analyzer_detect_sysroot.rust_analyzer_toolchain.json"
```

Comment:
Should this use `{workspace_name}` instead of `rules_rust` as the first element of this path? I assume the work was being done to set this up before for a reason?

Reply:
I think it was previously `workspace_name` to try to handle the case where `rules_rust` is imported under a different name. But the new runfiles API (`rlocation!`) knows how to read the repo_mapping file, so it would convert `rules_rust` to the remapped name, if necessary